// Copyright 2021 The Cross-Media Measurement Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package wfanet.panelmatch.protocol.protobuf;

option java_package = "wfanet.panelmatch.protocol.protobuf";
option java_multiple_files = true;

// The request to apply encryption.
message ApplyEncryptionRequest {
  repeated bytes plaintexts = 1;
  bytes encryption_key = 2;
}

// Response of the ApplyEncryptionRequest method.
message ApplyEncryptionResponse {
  repeated bytes encrypted_texts = 1;
  int64 elapsed_cpu_time_millis = 2;
}

// The request to reapply encryption.
message ReApplyEncryptionRequest {
  repeated bytes encrypted_texts = 1;
  bytes encryption_key = 2;
}

// Response of the ReApplyEncryptionRequest method.
message ReApplyEncryptionResponse {
  repeated bytes reencrypted_texts = 1;
  int64 elapsed_cpu_time_millis = 2;
}

// The request to apply decryption.
message ApplyDecryptionRequest {
  repeated bytes encrypted_texts = 1;
  bytes encryption_key = 2;
}

// Response of the ApplyEncryptionRequest method.
message ApplyDecryptionResponse {
  repeated bytes decrypted_texts = 1;
  int64 elapsed_cpu_time_millis = 2;
}
